tests/grid_test.py

```diff
@@ -22,125 +21,56 @@ def test_error_nested_basis_set_type():
-    test_nested_basis_set = ChebyshevFirstKind.make_nested_set(2)
-    test_nested_basis_set_ = ChebyshevFirstKind.make_nested_set(1)
-    test_class = SmolyakGridGenerator(test_nested_basis_set)
-    test_class_ = SmolyakGridGenerator(test_nested_basis_set_)
-    expected_integer_grid_points_one_dimension = [[0], [1], [2], [3], [4]]
-    expected_integer_grid_points_two_dimension = [[0, 0], [1, 0], [2, 0],
-                                                  [0, 1], [0, 2], [3, 0],
-                                                  [4, 0],
-                                                  [1, 1], [1, 2], [2, 1],
-                                                  [2, 2], [0, 3], [0, 4]]
-    expected_levels_one_dimension = [[0], [1], [1], [2], [2]]
-    expected_levels_two_dimension = [[0, 0], [1, 0], [1, 0], [0, 1], [0, 1],
-                                     [2, 0], [2, 0], [1, 1], [1, 1], [1, 1],
-                                     [1, 1], [0, 2], [0, 2]]
-    expected_grid_points_one_dimension = [[0], [-1], [1],
-                                          [-0.70710], [0.70710]]
-    expected_grid_points_two_dimension = [[0, 0], [-1, 0], [1, 0], [0, -1],
-                                          [0, 1],
-                                          [-0.70710, 0], [0.70710, 0],
-                                          [-1, -1],
-                                          [-1, 1], [1, -1], [1, 1],
-                                          [0, -0.70710],
-                                          [0, 0.70710]]
-    # evaluate grid points basis functions at test point in one dimension
-    test_point = 0.72
-    test_grid_points_basis_eval_one_dimension = [
-        grid_points_basis.__call__(test_point)
-        for grid in test_class(1).basis_functions
-        for grid_points_basis in grid]
-    expected_grid_points_basis_one_dimension = [1, 0.72, 2*(0.72**2)-1,
-                                                4*0.72**3-3*0.72,
-                                                8*0.72**4 - 8*0.72**2+1]
-    # evaluate grid points basis functions at test point in two dimension
-    test_grid_points_basis_eval_two_dimension = []
-    for grid_index in range(len(test_class_(2).basis_functions)):
-        test_grid_points_basis_eval_two_dimension.append([])
-        for grid_basis_function in test_class_(2).basis_functions[grid_index]:
-            test_grid_points_basis_eval_two_dimension[grid_index].append(
-                grid_basis_function.__call__(test_point))
-    expected_grid_points_basis_two_dimension = [[1, 1], [0.72, 1],
-                                                [2*(0.72**2)-1, 1],
-                                                [1, 0.72], [1, 2*(0.72**2)-1]]
+    grid_gen = SmolyakGridGenerator(ChebyshevFirstKind.make_nested_set(2))
+    grid = grid_gen(1)
+    assert grid.indexes == [0, 1, 2, 3, 4]
+    assert grid.levels == [0, 1, 1, 2, 2]
+    assert numpy.allclose(grid.points, [0, -1, 1, -0.70710, 0.70710], atol=1e-10)
```

Comment:
Is there a reason to make atol stricter than the default? Suggest to remove:
```suggestion
    assert numpy.allclose(grid.points, [0, -1, 1, -0.70710, 0.70710])
```